docs/_ext/autoclass_with_examples.py

```diff
@@ -0,0 +1,94 @@
+import re
+import sys
+from typing import Any, Dict, List, Tuple
+
+from docutils.nodes import Node
+from sphinx.ext.autodoc.directive import AutodocDirective
+
+from starknet_py.constants import ROOT_PATH
+
+
+class AutoclassWithExamples(AutodocDirective):
+    """
+    Directive class extending AutodocDirective and adding code examples.
+    """
+
+    def run(self) -> List[Node]:
+        original_class = getattr(
+            sys.modules[self.env.ref_context.get("py:module")], self.arguments[0]
+        )
+        add_code_examples(original_class)
+
+        self.name = self.name.replace("-with-examples", "")  # remove `-with-examples`
```

Comment:
Maybe we could make this a static method `get_class_name`?

Reply:
Not really, AutodocDirective is using it this way